nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/lifecycle/event/BulkLifeCycleChangeListener.java

```diff
@@ -140,21 +143,30 @@ protected boolean isNonRecursiveTransition(String transition, String type) {
         return nonRecursiveTransitions.contains(transition);
     }
 
-    protected void changeChildrenState(CoreSession session, DocumentRef parentRef, String transition,
-            String targetState) {
-        boolean paginate = Framework.getService(ConfigurationService.class).isBooleanPropertyTrue("blabla");
+    protected void changeChildrenState(CoreSession session, ConfigurationService confService, String transition,
+            String targetState, DocumentModel doc) {
```

Comment:
Pass the computed `pageSize` itself (and `-1` if no pagination) instead of passing the `ConfigurationService` and re-doing the lookup each time.